arbnode/batch_poster.go

```diff
@@ -369,6 +377,41 @@ func NewBatchPoster(ctx context.Context, opts *BatchPosterOpts) (*BatchPoster, e
 	return b, nil
 }
 
+type dummyBlobReader struct {
+	blobs []kzg4844.Blob
+}
+
+func (b *dummyBlobReader) GetBlobs(ctx context.Context, batchBlockHash common.Hash, versionedHashes []common.Hash) ([]kzg4844.Blob, error) {
+	return b.blobs, nil
+}
+func (b *dummyBlobReader) Initialize(ctx context.Context) error { return nil }
+
+type testMuxBackend struct {
```

Comment:
Minor: The name of this confused me a bit when I saw it, I thought it was for testing as in `_test.go`. If you can think of a different name and/or include a comment about what it's for that would be great.